tools/functional-tester/etcd-tester/checks.go

```diff
@@ -87,29 +89,93 @@ func (lc *leaseChecker) Check() error {
 		return err
 	}
 	plog.Infof("checking short lived leases %v", lc.ls.shortLivedLeases.leases)
-	return lc.check(true, lc.ls.shortLivedLeases.leases)
+	return lc.shortLivedLeaseCheck()
+
+}
+
+// shortLivedLeaseCheck verifies the expiry of shortLivedLeases.
+// the logic is follwing:
+//		1. for each lease from shortLivedLeases map
+//			1. get lease remaining TTL info.
+//			2. if lease is not expired, then wait until it expires.
+//			3. if lease is expired but not deleted, then wait for 1 second for etcd cluster to process lease deletion.
+//			4. check lease expiry.
+//				1. retry again by repeating steps 1 - 4 if check fails
+//				2. exit if check succeeds
+//			5. return error if all retries have failed
+//		2. return any errors on checking lease from shortLivedLeases
+func (lc *leaseChecker) shortLivedLeaseCheck() error {
+	ctx, cancel := context.WithTimeout(context.Background(), leaseCheckerTimeout)
+	errc := make(chan error)
+	defer cancel()
+	for leaseID := range lc.ls.shortLivedLeases.leases {
+		go func(id int64) {
+			var err error
+			// retry in case of transient failure or lease is expired but not deleted due to the fact that etcd cluster didn't have enought time to delete it.
+			for i := 0; i < retries; i++ {
+				resp, err := lc.ls.getLeaseByID(ctx, leaseID)
+				if rpctypes.Error(err) == rpctypes.ErrLeaseNotFound {
+					break
+				}
+				if err != nil {
+					plog.Warningf("retry %d. failed to retrieve lease %v error (%v)", i, id, err)
+				} else {
+					if resp.TTL > 0 {
+						// sleep for the lease remaining time (represented by resp.TTL)
+						plog.Warningf("lease %v is not expired. sleep for %d until it expires.", id, resp.TTL)
+						time.Sleep(time.Duration(resp.TTL) * time.Second)
+					} else {
+						// lease is expired but not deleted might due to the fact that there aren't enough time for etcd cluster to delete expired leases.
+						// we sleep for 1 second and retry again.
```

Comment:
this comment seems redundant; the two lines below already say "retry" and "Sleep(time.Second)"